activerecord/lib/active_record/counter_cache.rb

```diff
@@ -73,13 +90,32 @@ def reset_counters(id, *counters)
       #   # UPDATE posts
       #   #    SET comment_count = COALESCE(comment_count, 0) + 1
       #   #  WHERE id IN (10, 15)
+      #
+      #   # For the Posts with id of 10 and 15, increment the comment_count by 1
+      #   # and update the updated_at value for each counter.
+      #   Post.update_counters [10, 15], comment_count: 1, updated_at: true
+      #   # Executes the following SQL:
+      #   # UPDATE posts
+      #   #    SET comment_count = COALESCE(comment_count, 0) + 1,
+      #   #    "updated_at" = '2016-10-13T09:59:23-05:00'
+      #   #  WHERE id IN (10, 15)
       def update_counters(id, counters)
+        object = find(id)
+        timestamp_columns = resolve_timestamp_columns(object, counters.delete(:touch))
+
         updates = counters.map do |counter_name, value|
           operator = value < 0 ? "-" : "+"
           quoted_column = connection.quote_column_name(counter_name)
           "#{quoted_column} = COALESCE(#{quoted_column}, 0) #{operator} #{value.abs}"
         end
 
+        if timestamp_columns
+          Array(timestamp_columns).each do |column|
+            quoted_column = connection.quote_column_name(column.to_s)
+            updates << "#{quoted_column} = '#{DateTime.now}'"
+          end
+        end
```

Comment:
I'd write it as:

```ruby
touch_time = Time.now
timestamps = resolve_timestamp_columns(find(id), touch) # Note the expected KW arg use.
timestamps.map do |column|
  updates << "#{connection.quote_column_name(column.to_s)} = '#{touch_time}'"
end
```

For MySQL we track columns with millisecond precision if it's configured right, so I think we must capture the touch time. Also we needn't bother with `DateTime`, Active Record should typecast it properly besides we'd need a `require "date"` at the top of the file here.